services/accessanalyzer/pom.xml

```diff
@@ -56,5 +56,10 @@
             <artifactId>aws-json-protocol</artifactId>
             <version>${awsjavasdk.version}</version>
         </dependency>
+        <dependency>
```

Comment:
Instead of adding it to every service module, can we add it to this pom? https://github.com/aws/aws-sdk-java-v2/blob/master/services/pom.xml This way, if we need to change anything, say artifact ID, we can do it just in one place.

Reply:
I added a note to the description about it

> Given that codecatalyst does not use http-auth-aws this dependency was added to all the services but that one instead of just adding it to the services/pom.xml parent pom.

So, we can add it to the `services/pom.xml` and live with one warning of unused dependency for `codecatalyst`, or we can just add it to all of the services but `codecatalyst` as done here and avoid having any warnings, thoughts?

Reply:
My bad, should've read the description first 😅 Agreed we should add the dependency to each service module since not all services use them. The last thing we want is adding unused dependency since it'll increase JAR size.

Reply:
We need to update this template though https://github.com/aws/aws-sdk-java-v2/blob/master/services/new-service-template/pom.xml This is used to create new service modules. We need to figure out how to add this dependency conditionally.

Reply:
The template is already updated, we do not have at the moment a way to "opt-out" other than editing the resulting pom. I can take a look at that.

Reply:
Discussed offline, new services will be addressed in another PR

Reply:
Added on #4309